### PR TITLE
Refactor scan screen to support all certificates (EXPOSUREAPP-7686)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Adds navigation, new permission api

test and recovery are incomplete due missing screen/ repo extension 
vaccination works

merge after https://github.com/corona-warn-app/cwa-app-android/pull/3444

### CODE REVIEW

**Review by NataliaLemmerth — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/scan/DccQrCodeScanFragment.kt [hunk 1]
```diff
@@ -23,16 +23,32 @@ import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
...
     AutoInject {
 
     @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
-    private val viewModel: VaccinationQrCodeScanViewModel by cwaViewModels { viewModelFactory }
+    private val viewModel: DccQrCodeScanViewModel by cwaViewModels { viewModelFactory }
 
     private val binding: FragmentScanQrCodeBinding by viewBinding()
     private var showsPermissionDialog = false
 
+    val requestPermissionLauncher =
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/scan/DccQrCodeScanFragment.kt:36]: can be `private` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewFragment.kt [hunk 2]
```diff
@@ -81,11 +80,9 @@ class PersonOverviewFragment : Fragment(R.layout.person_overview_fragment), Auto
...
             }
 
-            ScanQrCode -> Toast.makeText(
-                requireContext(),
-                "TODO \uD83D\uDEA7 Tomorrow maybe?!",
-                Toast.LENGTH_LONG
-            ).show()
+            ScanQrCode -> doNavigate(
+                PersonOverviewFragmentDirections.actionPersonOverviewFragmentToDccQrCodeScanFragment()
+            )
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewFragment.kt:85]: Can we apply the same transition as in `CheckInsFragment` ?

**Review by mtwalli — COMMENTED** [review]
Looks nice, just small comments

**Review by d4rken — APPROVED** [review]
lgtm

**Review by mtwalli — APPROVED** [review]
Created a ticket for polishing https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7984

**Pull request merged**